src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs

```diff
@@ -86,6 +86,7 @@ public DirectoryAssemblyResolver (Action<TraceLevel, string> logger, bool loadDe
 			this.loadDebugSymbols = loadDebugSymbols;
 			this.logger       = logger;
 			SearchDirectories = new List<string> ();
+			this.resolutionPath = new List<AssemblyNameReference> ();
```

Comment:
***Alternate Take***: *Don't* add a new `AssemblyNotFoundException` type or otherwise change the Java.Interop repo. Instead, all behavior could be done within xamarin-android and the `<ResolveAssemblies/>` task, by using try/catch:

```csharp
void AddAssemblyReferences (DirectoryAssemblyResolver resolver, ICollection<string> assemblies, AssemblyDefinition assembly, bool topLevel, new List<AssemblyNameReference> resolutionPath /* new parameter! */)
{
    // ...
    LogMessage ("{0}Adding assembly reference for {1}, recursively...", new string (' ', indent), assembly.Name);
    indent += 2;
    resolutionPath.Add (assembly.Name);
    // ...
    foreach (AssemblyNameReference reference in assembly.MainModule.AssemblyReferences) {
        AssemblyDefinition reference_assembly;
        try {
            reference_assembly = resolver.Resolve (reference);
        }
        catch (FileNotFoundException e) {
            LogError (...);
            return;
        }
        AddAssemblyReferences (resolver, assemblies, reference_assembly, false, resolutionPath);
    }
    indent -= 2;
    resolutionPath.Remove (resolutionPath.Count-1);
}
```

This way, the change is localized to one repo (yay), and we're able to print out a *better* error message (yay), and it could be an *actual* error message instead of an "Unhandled exception" message (yay).